README.md

```diff
@@ -1,49 +1,72 @@
 # L1 -> L3 ERC20 Teleportation
 
-Contracts enabling direct ERC20 bridging from L1 to L3.
+Contracts enabling direct L1 to L3 ERC20 bridging. Teleportations are ERC20 deposits from L1 through any Arbitrum L2 to any Arbitrum L3 on the L2.
 
-### Summary
+## Summary
 
 In short, there are 3 steps to an L1 -> L3 teleportation:
 1. Send funds from L1 to a personal `L2Forwarder` whose address depends on its parameters
 2. Create the `L2Forwarder` if it doesn't already exist and start the third step
 3. Send tokens and ETH from the `L2Forwarder` to the recipient on L3
 
-### Deployment Procedure
-1. Deploy `L2ForwarderContractsDeployer` using a generic CREATE2 factory such as `0x4e59b44847b379578588920cA78FbF26c0B4956C` on each L2. This will deploy the `L2ForwarderFactory` and `L2Forwarder` implementation. Use the same salt on each L2.
-2. Deploy `L1Teleporter` to L1, passing the L2 factory and implementation addresses to the constructor.
+### Contracts
 
-### Teleportation Flow
+- `L1Teleporter`
+    - Deployed to Ethereum L1
+    - Users call `teleport()` to initiate "teleportations"
+    - Takes tokens from the user and sends them through an L2’s token bridge to a specific `L2Forwarder` whose address is predicted by CREATE2 logic
+    - Creates a retryable to create and call the `L2Forwarder` via the `L2ForwarderFactory`
+    - Can be paused by an owner(s)
+- `L2ForwarderFactory`
+    - Deployed to any Arbitrum L2 at the same address
+    - Creates, initializes and calls `L2Forwarder` clones
+- `L2Forwarder`
+    - Implementation deployed to all Arbitrum L2’s at the same address
+    - Factory-deployed clones receive funds from L1 and forward them to L3
 
-1. User approves `L1Teleporter` to spend TOKEN (if the L3 uses a custom fee token, approve the fee token too)
-2. User calls `L1Teleporter.teleport`:
-    1. Computes the `L2Forwarder` address
-    2. Sends tokens over the bridge to the `L2Forwarder`. (if using a custom fee token, send those to the forwarder as well)
-    3. Creates a retryable to call `L2ForwarderFactory.callForwarder`
-3. Token bridge retryable(s) redeemed: tokens land in the `L2Forwarder` address (which may or may not have code yet)
-4. Factory retryable redeemed: `L2ForwarderFactory.callForwarder`
-    1. Create and initialize the user's `L2Forwarder` via `Clone` if it does not already exist.
-    2. Call `L2Forwarder.bridgeToL3(...)`
-5. `L2Forwarder.bridgeToL3`
-    1. Send the forwarder's entire token balance through the bridge to L3. The contract's entire ETH (or fee token) balance minus execution fee is sent as submission fee in order to forward all the extra ETH (or fee token) to L3.
+## Deployment Procedure
+Deploy `L1Teleporter`  on any L1, passing in predicted `L2Forwarder`  and `L2ForwarderFactory`  arguments. Deploy `L2ForwarderContractsDeployer` using a generic CREATE2 factory to the same address on any Arbitrum L2’s.
 
-### Retryable Failures and Race Conditions
```

Comment:
why is this section removed?

Reply:
i just thought it was too much :shrug: 

added it back in

Reply:
kinda agree, how about add to a separate markdown file?

Reply:
sounds good to me